cmake/modules/dts.cmake

```
@@ -290,10 +290,16 @@ execute_process(
   COMMAND ${CMD_GEN_DEFINES}
   WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
   RESULT_VARIABLE ret
+  ERROR_VARIABLE stderr
   )
 if(NOT "${ret}" STREQUAL "0")
   message(STATUS "In: ${PROJECT_BINARY_DIR}, command: ${CMD_GEN_DEFINES}")
-  message(FATAL_ERROR "gen_defines.py failed with return code: ${ret}")
+  # If gen_defines failed with an error message, print it as part of the fatal error
+  if (stderr)
+    message(FATAL_ERROR "gen_defines.py failed due to ${stderr}")
+  else()
+    message(FATAL_ERROR "gen_defines.py failed with return code: ${ret}")
+  endif()
```

let's just print all knowledge we have whenever there is a failure.
```suggestion
  message(FATAL_ERROR "gen_defines.py failed with return code: ${ret}\nstderr: ${stderr}")
```

---

What do you think about the following?
It avoids printing the `stderr:\n` when there's nothing in `stderr` (which, arguably, should never happen...)
```suggestion
  if (stderr)
    set(stderr "\nScript stderr: ${stderr}")
  endif()
  message(FATAL_ERROR "gen_defines.py failed with return code: ${ret}${stderr}")
```

---

> It avoids printing the `stderr:\n` when there's nothing in `stderr` 

I actually think that a `stderr: ` with no content actually indicates to users that there actually was no error message, which can also be useful information.

If not, then users would actually need to know that `stderr` is only missing if it is empty, something they can only know if they look in the CMake file.

---

I see your point. However, I think an empty `stderr: ` could be confusing.
Would displaying `stderr: <empty>` instead be fine by you?

---

> Would displaying stderr: <empty> instead be fine by you?

yes, that's a clear indication to the user that an error ocured, but that there were no messages printed to stderr.